src/csharp/Microsoft.Spark/Utils/UdfSerDe.cs

```diff
@@ -184,30 +244,62 @@ private static TypeData SerializeType(Type type)
 
         private static Type DeserializeType(TypeData typeData) =>
             s_typeCache.GetOrAdd(typeData,
-                td => LoadAssembly(typeData.ManifestModuleName).GetType(typeData.Name));
+                td => LoadAssembly(td.AssemblyName, td.ManifestModuleName).GetType(td.Name));
+
+        /// <summary>
+        /// Return the cached assembly, otherwise attempt to load and cache the assembly
+        /// in the following order:
+        /// 1) Search the assemblies loaded in the current app domain.
+        /// 2) Load the assembly from disk using manifestModuleName.
```

Comment:
Is this same behavior as `AssemblyLoadContext`? If so, can you add a comment?

Reply:
https://docs.microsoft.com/en-us/dotnet/framework/deployment/best-practices-for-assembly-loading#avoid-loading-multiple-versions-of-an-assembly-into-the-same-context  suggests that we can use this method to check loaded assemblies.